### PR TITLE
add tests for sender, receiver open errors

### PR DESCRIPTION
**Pull request by ortex**
#67

### CODE REVIEW

**Comment by gavv** [comment]
We could nicely merge new tests into TestXxx_Open by adding context creation func to the table:

```go
		{
			name:    "closed context",
			...
			contextFunc: func() *Context {
				ctx, err := OpenContext()
				ctx.Close()
				return ctx
			},
		},
```

WDYT?

File: roc/receiver_test.go [hunk 1]
```diff
@@ -9,12 +9,17 @@ import (
...
-		wantErr error
+		name        string
+		contextFunc func() *Context
+		config      ReceiverConfig
+		wantErr     error
 	}{
 		{
 			name: "ok",
+			contextFunc: func() *Context {
+				ctx, _ := OpenContext(ContextConfig{})
```
Comment (gavv) [inline roc/receiver_test.go:20]: I'd suggest to check that error is nil here and in other context funcs.

File: roc/receiver_test.go [hunk 2]
```diff
@@ -9,12 +9,17 @@ import (
...
+				return ctx
+			},
 			config: ReceiverConfig{
 				FrameSampleRate: 44100,
 				FrameChannels:   ChannelSetStereo,
```
Comment (gavv) [inline roc/receiver_test.go:25]: Let's use makeReceiverConfig here?

File: roc/receiver_test.go (hunk 1 above)
Comment (gavv) [reply roc/receiver_test.go:20]: Same for sender

File: roc/receiver_test.go (hunk 2 above)
Comment (gavv) [reply roc/receiver_test.go:25]: Same for sender

File: roc/receiver_test.go (hunk 1 above)
Comment (ortex) [reply roc/receiver_test.go]: done

File: roc/receiver_test.go (hunk 2 above)
Comment (ortex) [reply roc/receiver_test.go]: agreed. Done

**Comment by gavv** [comment]
LGTM!

**Pull request merged**